<?xml version="1.0"?>

<launch>
	<arg name="host" default="10.0.0.17" />
	<arg name="port" default="41451" />

	<node name="airsim_node" pkg="airsim_ros_pkgs" type="airsim_node" output="log">
		<param name="is_vulkan" type="bool" value="false" />
		<param name="update_airsim_img_response_every_n_sec" type="double" value="0.5" />
		<param name="update_airsim_control_every_n_sec" type="double" value="0.1" />
		<param name="update_lidar_every_n_sec" type="double" value="0.5" />
		<param name="publish_clock" type="bool" value="false" />
		<param name="host_ip" type="string" value="$(arg host)" />
	</node>

	<!-- Static transforms -->
	<include file="$(find airsim_ros_pkgs)/launch/static_transforms.launch"/>

	<node name="global_planner" pkg="assignment_pkg" type="global_planner.py" output="screen">
   		<param name="host" type="string" value="$(arg host)" />
		<param name="port" type="int" value="$(arg port)" />
		<param name="weather" type="string" value="" />
		<param name="weather_value" type="double" value="0.25" />
   	</node>

	<node name="local_planner" pkg="assignment_pkg" type="local_planner.py" output="screen">
   		<param name="host" type="string" value="$(arg host)" />
		<param name="port" type="int" value="$(arg port)" />
   	</node>

	<node name="elevation_cli" pkg="assignment_pkg" type="elevation_cli.py" output="screen">
		<param name="host" type="string" value="$(arg host)" />
		<param name="port" type="int" value="$(arg port)" />
	</node>
   
	<node name="elevation_service" pkg="assignment_pkg" type="elevation_service.py" output="screen"/>
</launch>
